### ISSUE TITLE
Web component: Locale selector - list of options (locations and languages) are not identified by the screen reader. Only a "List box" is identified

### ISSUE THREAD

**Issue by Praveen-sr** [carbon-design-system/carbon-for-ibm-dotcom#4668]
## Detailed description

Issue:
The list of options are not identified by the screen reader. Only a "List box" is identified instead. Here within the list box, when we navigate with up/down arrow keys, Screen reader does not announce anything. The options are announced only in tab/shift tab navigation. This behavior would be confusing. Hence the user would not be able to interact with the elements efficiently.

Expected:
Remove the role "list box" from the options.
